DatadogSessionReplay/Sources/SessionReplay.swift

```diff
@@ -45,13 +78,46 @@ public enum SessionReplay {
         let resources = ResourcesFeature(core: core, configuration: configuration)
         try core.register(feature: resources)
 
-        let sessionReplay = try SessionReplayFeature(core: core, configuration: configuration)
+        let sessionReplay = try SessionReplayFeature(core: core, configuration: configuration, startRecordingImmediately: startRecordingImmediately)
         try core.register(feature: sessionReplay)
 
         core.telemetry.configuration(
             defaultPrivacyLevel: configuration.defaultPrivacyLevel.rawValue,
-            sessionReplaySampleRate: Int64(withNoOverflow: configuration.replaySampleRate)
+            sessionReplaySampleRate: Int64(withNoOverflow: configuration.replaySampleRate),
+            startRecordingImmediately: startRecordingImmediately
         )
     }
+
+    internal static func startRecording(core: DatadogCoreProtocol) throws {
+        guard !(core is NOPDatadogCore) else {
+            throw ProgrammerError(
+                description: "Datadog SDK must be initialized before calling `SessionReplay.startRecording()`."
+            )
+        }
+
+        guard let sr = core.feature(named: SessionReplayFeaturneName, type: SessionReplayFeature.self) else {
```

Comment:
nit/ It might be safer to get it by type: `core.get(feature: SessionReplayFeature.self)`